### PR TITLE
Add convenience field support to Terraform, generate Network

### PR DESCRIPTION
**Pull request by rileykarson**
Sorry for the # of changes! A few things had to happen for this PR.

Contained, roughly in order is;

- Add overrides for Terraform
- Add support for flattening objects
- Fix broken TF tests, test the actual complex cases (updating, legacy networks)
- Fix update for Network in `api.yaml`
- Clarify docs, remove the old resource-level message because it was outdated

Going from my definition of convenience fields;

> `convenience fields`: When we take a nested object field and top-level it.
> useful for nested objects with non-zero defaults.
> example: BackendService connection_draining.draining_timeout_sec -> connection_draining_timeout_sec

This is adding support for the most useful subset, when a nested objects has been replaced entirely. Most? examples of convenience fields are a single field inside of an object, and this solution will unblock those cases. Support for nested convenience fields is a TODO, and something we should do if it's necessary, not just for fun.

I looked at using the overrides system to do this, and specifying `convenience_name` or something similar on the convenience field itself, but that approach didn't work well in the end. Our model for resource schemas relies pretty heavily on 1:1 mappings API <-> tool, and there are a lot of yaks to shave before that could be changed.

-----------------------------------------------------------------
# [all]
## [terraform]
Generate google_compute_network using Magic Modules

### CODE REVIEW

File: templates/terraform/objectlib/base.go.erb [hunk 1]
```diff
@@ -13,6 +13,18 @@ package google
 func Get<%= resource_name -%>ApiObject(d TerraformResourceData, config *Config) (map[string]interface{}, error) {
     obj := make(map[string]interface{})
 <%  object.settable_properties.each do |prop| -%>
+<%    if prop.flatten_object -%>
+    <%= prop.api_name -%>Prop, err := expand<%= resource_name -%><%= titlelize_property(prop) -%>(d, config)
+    if err != nil {
+        return err
```
Comment (ndmckinley) [inline templates/terraform/objectlib/base.go.erb]: Should be "return nil, err", I think?

**Comment by ndmckinley** [comment]
This one is a challenge to review without the downstreams - I'll see if I can make sense of them when they come out.

**Comment by ndmckinley** [comment]
If the network tests pass (and once you fix the quotes that travis is complaining about, and two minor tweaks on the beta / ga modules), I think I trust this to be right.

File: api/type.rb [hunk 2]
```diff
@@ -26,6 +26,11 @@ module Fields
       attr_reader :description
       attr_reader :exclude
 
+      # Add a deprecation message for a field that's been deprecated in the API
+      # use the YAML chomping folding indicator (>-) if this is a multiline
+      # string, as providers expect a single-line one w/o a newline.
+      attr_reader :deprecation_message
```
Comment (rileykarson) [inline api/type.rb]: fyi @rambleraptor

**Pull request merged**